bin/src/extract.rs

```diff
@@ -79,8 +79,26 @@ pub struct Options {
 #[derive(Debug, Args)]
 pub struct Target {
     /// Image reference being extracted (e.g. docker.io/library/ubuntu:latest)
-    #[arg(value_parser = Reference::from_str)]
-    pub image: Reference,
+    ///
+    /// If a fully specified reference is not provided,
+    /// the image is attempted to be resolved with the prefix
+    /// `docker.io/library`.
+    ///
+    /// The reference may optionally provide a digest, for example
+    /// `docker.io/library/ubuntu@sha256:1234567890`.
+    ///
+    /// Finally, the reference may optionally provide a tag, for example
+    /// `docker.io/library/ubuntu:latest` or `docker.io/library/ubuntu:24.04`.
+    /// If no digest or tag is provided, the tag "latest" is used.
+    ///
+    /// Put all that together and you get the following examples:
+    /// - `ubuntu` is resolved as `docker.io/library/ubuntu:latest`
+    /// - `ubuntu:24.04` is resolved as `docker.io/library/ubuntu:24.04`
+    /// - `docker.io/library/ubuntu` is resolved as `docker.io/library/ubuntu:latest`
+    /// - `docker.io/library/ubuntu@sha256:1234567890` is resolved as `docker.io/library/ubuntu@sha256:1234567890`
+    /// - `docker.io/library/ubuntu:24.04` is resolved as `docker.io/library/ubuntu:24.04`
+    #[arg(verbatim_doc_comment)]
+    pub image: String,
```

Comment:
LGTM. 

[Optional, probably more relevant for the CLI and super minor] It might be worth making the "base" configurable via env var. If I'm a company who has a list of image names it may be simpler for me to set an env var specifying the base rather than have to program my CI jobs to concatenate them together. It also means that I can create a CI job template that sets the var and makes makes calls to this do the right thing rather than making my engineers have to remember to use the FQN. 

^^ I'm a little dubious that this is _super_ useful, but I think it's worth explicitly rejecting.

Reply:
that's an incredible idea! made that quick change.

Reply:
I was thinking about this more over lunch, and it would be similar to also have CLI options that can do that.

Reply:
IMO this is behavior that we probably _don't_ want in flags; it's not really intended to be used most of the time, it's just an escape hatch.

Reply:
Makes sense, thanks!